options/internal_flash.cpp

```diff
@@ -17,6 +17,14 @@
 
 #include "FlashIAPBlockDevice.h"
 
+// Assert that the user has specified a flash region
+#define INTERNAL_FLASH true
+#if MBED_CONF_STORAGE_SELECTOR_STORAGE
+MBED_STATIC_ASSERT(MBED_CONF_FLASHIAP_BLOCK_DEVICE_SIZE > 0,
+    "No flash region specified, flashiap-block-device.size is 0");
```

Comment:
Maybe give a suggestion on how to fix the problem? Something like: "No flash region specified. Set configuration option "flashiap-block-device.size" to an integer greater than 0."

Also, will this only be asserted if INTERNAL_FLASH is selected? It seems like this would always be asserted.

Reply:
> I think the code change speaks for itself.

I'm with @bridadan - it looks like the code will assert even when choosing other storage devices?

Reply:
I like Brian speak he speak good.

It's unfortunately not straightforward (I had to update after a CI failure and now the code no longer speaks). But the `#define INTERNAL_FLASH true` and `#if MBED_CONF_STORAGE_SELECTOR_STORAGE` are only true if storage-selector.storage is set to INTERNAL_FLASH.

I couldn't figure out how to make it prettier, but I'm open to suggestions.

Reply:
I went through it with @geky and it the conditional-ness makes sense. It's crazy like he said, but I _think_ it should be fine.

Reply:
Let's ship it then! 😄